README.md

```diff
@@ -6,46 +6,286 @@
+Fast forward merge is just equivalent to moving tip of a branch to which you merge to the same commit as a tip of a merged branch.
+This is what `Fast Forward Parent Branch To Match Current Branch` context menu action does.
+As you can see, the edge between a given branch and its parent was changed to grey &mdash; it means that the child branch has been merged.
+
+![](docs/fast_forward.gif)
+</details>
+
+<details>
+<summary><strong>Slide out branch</strong></summary>
+<br/>
+
+When a branch was merged (and sometimes in other cases as well), you probably want to remove this branch from branch layout.
+To do this with ease (and without manual editing of `machete` file) you can use `Slide Out Selected Branch` action.
+This action will modify the branch layout, removing the entry for the selected branch `X` and reattaching `X`'s children as new children of `X`'s parent.
+If `machete.slideOut.deleteLocalBranch` git configuration key is set to `true`, then this action will also delete `X` branch from the underlying git repository (and not just from `.git/machete` file).
+The easiest way to set this key is using `git config --add machete.slideOut.deleteLocalBranch true` command inside a given repository.
+If you want to set this key globally (to be used for all repositories that don't have this key set explicitly), add `--global` option to the previous command.
+
+![](docs/slide_out_leaf_node.gif)
+
+![](docs/slide_out_non_leaf_node.gif)
+</details>
 
-### Prerequisites
+<details>
+<summary><strong>Slide in branch</strong></summary>
+<br/>
+
+You can also create a new branch below the selected one (also without editing `machete` file manually).
+To do this, choose the `Slide In Branch Below Selected Branch` action from a context menu.
+Type the name of new branch in the dialog window and click `Slide In` button.
+Another dialog will appear &mdash; it is a standard new branch IntelliJ dialog.
+
+![](docs/slide_in_new_branch.gif)
+
+A slide in action can also add a branch that already exists in git repository to the branch layout.
+Again, select the `Slide In Branch Bellow Selected Branch` from a context menu and type the name of an already existing branch.
+
+![](docs/slide_in_existent_branch.gif)
+
+This action can also be used to reattach an existing branch below a selected one.
+In this case, you just must type a name of the existing branch that you want to reattach and then click `Slide In`.
+You can also select the `Reattach children` checkbox to move along all children of the reattached branch.
+
+![](docs/slide_in_reattach_without_children.gif)
+
+![](docs/slide_in_reattach_with_children.gif)
+</details>
+
+<details>
+<summary><strong>Override fork point</strong></summary>
+<br/>
+
+According to [git documentation](https://git-scm.com/docs/git-merge-base#Documentation/git-merge-base.txt---fork-point),
+fork point is `the point at which a branch (or any history that leads to <commit>) forked from another branch (or any reference) <ref>`.
```

Comment:
Not sure if this one sentence is enough but probably with sentences below it is.

Reply:
Okay! just remember that we define fork point slightly differently than the `git merge-base`: